tensorboard/components/tf_scalar_dashboard/tf-scalar-dashboard.html

```diff
@@ -100,165 +94,188 @@
-      }
-
-      .log-button-selected {
-        background: var(--tb-ui-light-accent);
-      }
-
       #tooltip-sorting {
         display: flex;
         font-size: 14px;
         margin-top: 5px;
       }
-
       #tooltip-sorting-label {
         margin-top: 13px;
-        margin-left: 28px;
       }
-
       #tooltip-sorting paper-dropdown-menu {
         margin-left: 10px;
         --paper-input-container-focus-color: var(--tb-orange-strong);
         width: 105px;
       }
+      #x-type-selector paper-button {
+        margin: 5px 3px;
+      }
       .line-item {
         display: block;
         padding-top: 5px;
       }
-
-      .sidebar-section {
-        border-top: solid 1px rgba(0, 0, 0, 0.12);
-        padding: 20px 0px 20px 30px;
+      .no-data-warning {
+        max-width: 540px;
+        margin: 80px auto 0 auto;
       }
     </style>
   </template>
 
   <script>
-    import {DashboardBehavior} from "../tf-dashboard-common/dashboard-behavior";
-    import {ReloadBehavior} from "../tf-dashboard-common/reload-behavior";
-    import {BackendBehavior} from "../tf-backend/behavior";
+    import {RequestManager} from "../tf-backend/requestManager";
+    import {getRuns, getTags} from "../tf-backend/backend";
     import {getRouter} from "../tf-backend/router";
-    import {runsColorScale} from "../tf-color-scale/colorScale";
     import * as storage from "../tf-storage/storage";
 
     Polymer({
       is: "tf-scalar-dashboard",
-      behaviors: [
-        DashboardBehavior("scalars"),
-        ReloadBehavior("tf-chart-scaffold"),
-        BackendBehavior,
-      ],
       properties: {
-        backend: Object,
-        dataType: {
-          type: String,
-          value: "scalar"
-        },
-        scalarUrl: {
-          type: Function,
-          value: function() {
-            return getRouter().pluginRunTagRoute('scalars', '/scalars');
-          },
-        },
         _showDownloadLinks: {
           type: Boolean,
           notify: true,
           value: storage.getBooleanInitializer('_showDownloadLinks', false, true),
-          observer: '_showDownloadLinksObserver'
+          observer: '_showDownloadLinksObserver',
         },
         _smoothingWeight: {
           type: Number,
           notify: true,
           value: storage.getNumberInitializer('_smoothingWeight', 0.6),
-          observer: '_smoothingWeightObserver'
+          observer: '_smoothingWeightObserver',
         },
         _smoothingEnabled: {
           type: Boolean,
-          computed: '_computeSmoothingEnabled(_smoothingWeight)'
+          computed: '_computeSmoothingEnabled(_smoothingWeight)',
         },
         _ignoreYOutliers: {
           type: Boolean,
           value: storage.getBooleanInitializer('_ignoreYOutliers', true, true),
           observer: '_ignoreYOutliersObserver',
         },
         _xType: {
-          type: String,
-          value: "step"
+          type: String,  // ["step", "relative", "wall"]
```

Comment:
Closure Compiler *should* be able to check that at compile-time, but we currently have it configured in a very relaxed mode. We'd like to make it stricter in the future. However I would argue that this code is still worth having regardless, because it aids in readability, and I'm pretty sure our settings will cause it to be optimized away.